File: FixFlow/FixFlow.Application/DTOs/Request/ConfirmPaymentRequest.cs
using System.ComponentModel.DataAnnotations;

namespace FixFlow.Application.DTOs.Request;

public class ConfirmPaymentRequest
{
    [Required(ErrorMessage = "Booking je obavezan.")]
    public int BookingId { get; set; }

    [Required(ErrorMessage = "Stripe PaymentIntent ID je obavezan.")]
    public string PaymentIntentId { get; set; } = string.Empty;
}

File: packages/fixflow_core/lib/models/requests/confirm_payment_request.dart
class ConfirmPaymentRequest {
  final int bookingId;
  final String paymentIntentId;

  ConfirmPaymentRequest({
    required this.bookingId,
    required this.paymentIntentId,
  });

  Map<String, dynamic> toJson() => {
        'bookingId': bookingId,
        'paymentIntentId': paymentIntentId,
      };
}
